### PR TITLE
Added support for `async def`.

### PR DESCRIPTION
**Pull request by abingham**
This includes a basic unit test, though it might be good to extend all of the
current function rename tests to test async as well.

### CODE REVIEW

**Comment by soupytwist** [comment]
Thank you for creating a fix. Can you please add to the test case: `@testutils.only_for('3.5')` (ropetest.testutils).

**Comment by emacsway** [comment]
Oh, I see that you've created the complete fix. Let me to think a little about the fix. Thank you, @abingham

File: rope/base/pyobjectsdef.py [hunk 1]
```diff
@@ -354,6 +354,9 @@ def _eval(type_=type_, arg=arg):
             self.names[node.name] = pynames.DefinedName(pyfunction)
         self.defineds.append(pyfunction)
 
+    def _AsyncFunctionDef(self, node):
+        return self._FunctionDef(node)
```
Comment (emacsway) [inline rope/base/pyobjectsdef.py]: This looks ok.

> An async def function definition. Has the same fields as FunctionDef

http://greentreesnakes.readthedocs.io/en/latest/nodes.html?highlight=AsyncFunctionDef#async-and-await

**Comment by emacsway** [comment]
I have no objections. Looks good. Please, don't forget add @testutils.only_for('3.5') to fix tests for Python2, as it was said by @soupytwist .

And we still need add support for Await, AsyncFor, AsyncWith, and YieldFrom. But it's not related to this ticket.

**Comment by abingham** [comment]
I've added the version check decorator. I'll also see about putting together a second PR with support for the other async bits.

**Comment by abingham** [comment]
@emacsway #218 includes support for the rest of the async nodes.

**Comment by soupytwist** [comment]
Thank you @abingham, sorry for the delay in closing this.

**Comment by abingham** [comment]
No worries. Glad I could pitch in!

**Pull request merged**